source/common/config/config_provider_impl.cc

```diff
@@ -92,10 +92,19 @@ void DeltaConfigSubscriptionInstance::applyDeltaConfigUpdate(
   // needed. Such logic could be generalized as part of this framework such that this function owns
   // the diffing and issues the corresponding call to add/modify/remove a resource according to a
   // vector of functions passed by the caller.
-  auto* typed_provider =
-      static_cast<DeltaMutableConfigProviderBase*>(getAnyBoundMutableConfigProvider());
-  ConfigSharedPtr config = typed_provider->getConfig();
-  typed_provider->onConfigUpdate([config, update_fn]() { update_fn(config); });
+  // For now each config provider has its own copy of config, we need to propagate the update to
+  // every provider.
+  for (auto* provider : mutable_config_providers_) {
```

Comment:
This makes sense to keep things simpler initially. With that said, the end goal is to share the config _for each worker thread_ across providers, so that only `N = # threads` configs need to be allocated, as opposed to `N * # providers`. Can you add a TODO to make sure this is documented?

Reply:
done.